index.js

```diff
@@ -0,0 +1,145 @@
+const URL = "https://ada-backtrek-api.herokuapp.com/trips";
+
+const reportStatus = function reportStatus(message) {
+  $('#status').html(message);
+};
+
+const reportError = (message, errors) => {
+  let content = `<p>${message}</p><ul>`;
+  for (const field in errors) {
+    for (const problem of errors[field]) {
+      content += `<li>${field}: ${problem}</li>`;
+    }
+  }
+  content += "</ul>";
+  reportStatus(content);
+};
+
+const loadTrips = function loadTrips() {
+  reportStatus("Loading trips... be patient");
+
+  const allTrips = $("#all-trips");
+  const tripList = $("#trip-list");
+
+  allTrips.remove();
+  tripList.empty();
+
+  axios.get(URL)
+    .then((response) => {
+      tripList.before("<h2 id='all-trips'>All Trips</h2>");
+      response.data.forEach((trip) => {
+        tripList.append(`<li id=${trip.id}>${trip.name}</li>`);
+      });
+      reportStatus(`Successfully loaded ${response.data.length} trips`);
+    })
+    .catch((error) => {
+      console.log(error.response);
+      if (error.response.data && error.response.data.errors) {
+        reportError(
+          `Encountered an error: ${error.message}`,
+          error.response.data.errors
+        );
+      } else {
+        reportStatus(`Encountered an error while loading trips: ${error.message}`);
+      }
+    });
+};
+
+const loadTripDetails = function loadTripDetails() {
+
+  const tripDetails = $("#trip-details");
+  tripDetails.empty();
+
+  axios.get(URL + `/${this.id}`)
+    .then((response) => {
+      const trip = response.data;
+      let html = "";
+
+      html += `<h2>Trip Details</h2>
+               <div>
+                 <h3>Name: ${this.innerHTML}<h3>
```

Comment:
It might make sense to break the code to render trip details out into a separate function - as is, this function is doing two things (making the request, and rendering data). See also http://callbackhell.com/